bundles/org.openhab.binding.somneo/src/main/java/org/openhab/binding/somneo/internal/SomneoHandler.java

```
@@ -0,0 +1,542 @@
+
+        super.dispose();
+    }
+
+    private void initConnector() {
+        if (connector == null) {
+            SomneoConfiguration config = getConfigAs(SomneoConfiguration.class);
+            HttpClient httpClient;
+            if (config.ignoreSSLErrors) {
+                logger.info("Using the insecure client for thing '{}'.", thing.getUID());
+                httpClient = httpClientProvider.getInsecureClient();
+            } else {
+                logger.info("Using the secure client for thing '{}'.", thing.getUID());
+                httpClient = httpClientProvider.getSecureClient();
+            }
+
+            connector = new SomneoHttpConnector(config, httpClient);
+        }
+    }
+
+    private void updateThingProperties() {
+        final SomneoHttpConnector connector = this.connector;
+        if (connector == null) {
+            return;
+        }
+
+        Map<String, String> properties = editProperties();
+        properties.put(Thing.PROPERTY_VENDOR, PROPERTY_VENDOR_NAME);
+
+        try {
+            final DeviceData deviceData = connector.fetchDeviceData();
+            String value = deviceData.getModelId();
+            if (value != null) {
+                properties.put(Thing.PROPERTY_MODEL_ID, value);
+            }
+            value = deviceData.getSerial();
+            if (value != null) {
+                properties.put(Thing.PROPERTY_SERIAL_NUMBER, value);
+            }
+
+            final WifiData wifiData = connector.fetchWifiData();
+            value = wifiData.getMacAddress();
+            if (value != null) {
+                properties.put(Thing.PROPERTY_MAC_ADDRESS, value);
+            }
+
+            final FirmwareData firmwareData = connector.fetchFirmwareData();
+            value = firmwareData.getVersion();
+            if (value != null) {
+                properties.put(Thing.PROPERTY_FIRMWARE_VERSION, value);
+            }
+
+            updateProperties(properties);
+        } catch (Exception e) {
```

Which exception types are expected? You should probably not catch all exceptions here.